src/bluesky/run_engine.py

```diff
@@ -2338,6 +2344,13 @@ async def _wait(self, msg):
         else:
             group = msg.kwargs["group"]
             move_on = msg.kwargs.get("move_on", False)
+
+        if isinstance(group, collections.abc.Iterable) and not isinstance(group, str):
+            status_objs = group
+            group = "__status_group__"
```

Comment:
because we use the same key every time, if we "move on" and then wait on a different set, will they effectively get merged?

We should probably add the `hash(tuple(group))` to the name or if status objects are not hashable (or we worry there might be non-hashable subclasses in the future...I don't remember off the top of my head...), `hash(tuple(sorted(id(s) for s in group)))`. 

Although using `id()` is a Bad Idea™️ , I think this is a reasonable use case as object identity is the identity we care about on `Status` objects.  

The sorting is so that `group=[sA, sB]` and `group=[sB, sA]` are identified as the same, probably want to do the same thing with teh hash as well....

Reply:
I see, I understand. I made the change to use id() to make a unique hash for the group as suggested.